store/store_kv.go

```diff
@@ -84,20 +88,32 @@ func (kv *kvStore) Unlock(storeLock *Lock) error {
 }
 
 func (kv *kvStore) getFullLockPath(key string) string {
-	return kv.lockPrefix + "/" + key
+	return kv.storeName + "/"+ "locks"+ "/" + key
+}
+
+func (kv *kvStore) getFullKey(key string) string {
+	// Legacy keys are created directly under the pwx/<cluster-id> key
+	// New keys will be created under pwx/<cluster-id>/<store-name>
+	if kv.legacy {
+		return key
+	}
+	return kv.storeName + "/" + key
 }
 
 func (kv *kvStore) LockWithKey(owner, key string) (*Lock, error) {
-	fullPath := kv.getFullLockPath(key)
-	kvPair, err := kv.lockWithKeyHelper(owner, fullPath)
+	kvPair, err := kv.lockWithKeyHelper(owner, key, true)
 	if err != nil {
 		return nil, err
 	}
+
 	kvPair.lockedWithKey = true
 	return kvPair, err
 }
 
-func (kv *kvStore) lockWithKeyHelper(owner, key string) (*Lock, error) {
+func (kv *kvStore) lockWithKeyHelper(owner, key string, lockWithkey bool) (*Lock, error) {
+	if lockWithkey {
```

Comment:
nit: can we just move this logic of creating a path to individual callers, so that it is explicit what key they need to lock on?  Similar to how you have in Lock() API. 
This makes it easier to understand what key will eventually be locked

Reply:
Addressed